.changeset/fuzzy-pens-work.md

```diff
@@ -0,0 +1,8 @@
+---
+"simple-stack-stream": minor
+---
+
+Modified how headers are passed to the `Response` constructor within the Astro middleware to ensure compatibility and proper operation
+
+- Before: The response was created by passing `response.headers` directly as the second argument to `new Response(render(), response.headers)`, which was incorrect according to the expected signature of the `Response` constructor.
+- After: The issue was corrected by wrapping `response.headers` in an object under the `headers` key, using the correct syntax `new Response(render(), { headers: response.headers })`. This ensures that headers are properly handled in the response generated by the middleware.
```

Comment:
```suggestion
Fix missing `text/html` header for Astro pages when using the simple-stack-stream middleware.
```

Really appreciate the thorough engineering notes! But I'm going to redraft to a simpler description of the fix users should expect. Want to use changelogs to describe the user-facing effect, rather than implementation. I'll add that to our contributing guide as well.